src/components/c-modal.vue

```diff
@@ -0,0 +1,410 @@
+               ref="stickyFooter"
+               :class="b('sticky-footer')"
+          >
+            <slot name="stickyFooter"></slot>
+          </div>
+        </div>
+      </div>
+    </dialog>
+  </Transition>
+</template>
+
+<script lang="ts">
+  import { enableBodyScroll, disableBodyScroll } from 'body-scroll-lock';
+  import { defineComponent, ref, Ref } from 'vue';
+  import propScale from '@/helpers/prop.scale';
+  import { IModifiers } from '@/plugins/vue-bem-cn/src/globals';
+
+  interface ISetup {
+    container: Ref<HTMLDivElement>;
+  }
+
+  /**
+   * Renders a modal dialog.
+   *
+   * Based on https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog.
+   */
+  export default defineComponent({
+    name: 'c-modal',
+
+    // components: {},
+
+    props: {
+      /**
+       * Toggles the visibility of the modal.
+       */
+      isOpen: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Allows defining a title.
+       */
+      title: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows defining whether the modal is closable.
+       */
+      isClosable: {
+        type: Boolean,
+        default: true,
+      },
+
+      /**
+       * Allows enabling/disabling the close button.
+       */
+      hasCloseButton: {
+        type: Boolean,
+        default: true,
+      },
+
+      /**
+       * Allows enabling/disabling close on outside click.
+       */
+      closeOnOutsideClick: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Allows modifying the size of the modal.
+       *
+       * Valid values: `[600, 700, 800]`
+       */
+      size: propScale(600, [
+        600,
+        700,
+        800,
+      ]),
+
+      /**
+       * Allows modifying the inner spacing of the modal.
+       *
+       * Valid values: `[0, 300, 500]`
+       */
+      spacing: propScale(500, [
+        0,
+        300,
+        500,
+      ]),
+    },
+    emits: [
+      'update:isOpen',
+      'open',
+      'close',
+    ],
+
+    setup(): ISetup {
+      const container = ref();
+
+      return {
+        container,
+      };
+    },
+    // data(): IData {
+    //   return {};
+    // },
+
+    computed: {
+      /**
+       * Returns modifier classes.
+       */
+      modifiers(): IModifiers {
+        return {
+          hasCloseButton: this.isClosable && this.hasCloseButton,
+          size: this.size,
+          spacing: this.spacing,
+        };
+      },
+    },
+    watch: {
+      isOpen: {
+        /**
+         * Triggers opening/closing modal.
+         */
+        handler(state): void {
+          if (state) {
+            this.open();
+          } else {
+            this.close();
+          }
+        },
+      },
+    },
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    mounted() {
+      if (this.isOpen) {
+        this.open();
+      }
+    },
```

Comment:
suggestion: Vue3 introduced watchEffect() method will only execute the callback function when this component mounts and also each time the count or count2 variable changes. It has same effect as watch() but it is not * **lazy** by default like the watch() method as described by multiple articles in the web. 

*lazy only executes the callback function when the source value has changed.

Probably the mounted() hook could be removed with using watchEffect()

**Source:** https://www.webmound.com/watch-vs-watcheffect-vue-composition-api/

Reply:
Thank you for the input, I was not aware of `watchEffect`. I think this can be really useful in some places.
But for this case here I think the `mounted`-hook is the correct place.
This code part is for cases, where the modal has to be opened initially, so we don't want to watch for changes, just checking on mounting if it has to be opened directly.

Reply:
@MB-web You are correct. If we need to run a check during the initial render, it's recommended to use the mounted hook instead. watchEffect only listens on state changes during the mount but as you said, we do not watch for changes.